pom.xml

```diff
@@ -199,6 +202,13 @@
             <artifactId>jackson-datatype-jsr310</artifactId>
             <version>${jackson.version}</version>
         </dependency>
+
+        <dependency>
+            <groupId>jakarta.servlet</groupId>
+            <artifactId>jakarta.servlet-api</artifactId>
+            <version>${jakarta.servlet.version}</version>
+            <scope>provided</scope>
```

Comment:
Not sure if we require the `provided` scope?

Reply:
I think `provided` here will only use the `jakarta.servlet` during compilation and test phases, and the final bundled application will rely on the runtime for the dependency? Maybe this is the preferred approach at CH, not sure.

Reply:
Is this not being included in spring-boot anyway? Do we need this explicit dependency at all?

Reply:
@rafahop I could not find any resource to verify if this dependency is now part of spring boot. I have tried to remove this dependency from POM but it leaves unresolved errors in IDE. I think it is not part of Spring boot.

Reply:
https://mvnrepository.com/artifact/jakarta.servlet/jakarta.servlet-api/usages

Reply:
I'd suggest using `mvn dependency:tree` to confirm

Reply:
@rafahop It looks an independent dependency 
```
[INFO] uk.gov.companieshouse:ocr-api:jar:unversioned
[INFO] +- org.springframework.boot:spring-boot-starter:jar:3.3.1:compile
[INFO] |  +- org.springframework.boot:spring-boot:jar:3.3.1:compile
[INFO] |  |  \- org.springframework:spring-context:jar:6.1.10:compile
[INFO] |  +- org.springframework.boot:spring-boot-autoconfigure:jar:3.3.1:compile
[INFO] |  +- org.springframework.boot:spring-boot-starter-logging:jar:3.3.1:compile
[INFO] |  |  +- ch.qos.logback:logback-classic:jar:1.5.6:compile
[INFO] |  |  |  \- ch.qos.logback:logback-core:jar:1.5.6:compile
[INFO] |  |  +- org.apache.logging.log4j:log4j-to-slf4j:jar:2.23.1:compile
[INFO] |  |  |  \- org.apache.logging.log4j:log4j-api:jar:2.23.1:compile
[INFO] |  |  \- org.slf4j:jul-to-slf4j:jar:2.0.13:compile
[INFO] |  +- jakarta.annotation:jakarta.annotation-api:jar:2.1.1:compile
[INFO] |  +- org.springframework:spring-core:jar:6.1.10:compile
[INFO] |  |  \- org.springframework:spring-jcl:jar:6.1.10:compile
[INFO] |  \- org.yaml:snakeyaml:jar:2.2:compile
[INFO] +- org.springframework.boot:spring-boot-starter-web:jar:3.3.1:compile
[INFO] |  +- org.springframework.boot:spring-boot-starter-json:jar:3.3.1:compile
[INFO] |  +- org.springframework.boot:spring-boot-starter-tomcat:jar:3.3.1:compile
[INFO] |  +- org.springframework:spring-web:jar:6.1.10:compile
[INFO] |  |  +- org.springframework:spring-beans:jar:6.1.10:compile
[INFO] |  |  \- io.micrometer:micrometer-observation:jar:1.13.1:compile
[INFO] |  |     \- io.micrometer:micrometer-commons:jar:1.13.1:compile
[INFO] |  \- org.springframework:spring-webmvc:jar:6.1.10:compile
[INFO] |     +- org.springframework:spring-aop:jar:6.1.10:compile
[INFO] |     \- org.springframework:spring-expression:jar:6.1.10:compile
[INFO] +- org.springframework.boot:spring-boot-starter-validation:jar:3.3.1:compile
[INFO] |  \- org.hibernate.validator:hibernate-validator:jar:8.0.1.Final:compile
[INFO] |     +- jakarta.validation:jakarta.validation-api:jar:3.0.2:compile
[INFO] |     +- org.jboss.logging:jboss-logging:jar:3.5.3.Final:compile
[INFO] |     \- com.fasterxml:classmate:jar:1.7.0:compile
[INFO] +- org.json:json:jar:20230227:compile
[INFO] +- net.sourceforge.tess4j:tess4j:jar:5.8.0:compile
[INFO] |  +- net.java.dev.jna:jna:jar:5.13.0:compile
[INFO] |  +- com.github.jai-imageio:jai-imageio-core:jar:1.4.0:compile
[INFO] |  +- org.apache.pdfbox:pdfbox:jar:2.0.29:compile
[INFO] |  |  +- org.apache.pdfbox:fontbox:jar:2.0.29:compile
[INFO] |  |  \- commons-logging:commons-logging:jar:1.2:compile
[INFO] |  +- org.apache.pdfbox:pdfbox-tools:jar:2.0.29:compile
[INFO] |  |  \- org.apache.pdfbox:pdfbox-debugger:jar:2.0.29:compile
[INFO] |  +- org.apache.pdfbox:jbig2-imageio:jar:3.0.4:compile
[INFO] |  +- commons-io:commons-io:jar:2.13.0:compile
[INFO] |  +- net.sourceforge.lept4j:lept4j:jar:1.18.1:compile
[INFO] |  +- org.jboss:jboss-vfs:jar:3.2.17.Final:compile
[INFO] |  \- org.slf4j:slf4j-api:jar:2.0.13:compile
[INFO] +- uk.gov.companieshouse:structured-logging:jar:3.0.8:compile
[INFO] |  \- org.slf4j:slf4j-simple:jar:2.0.13:compile
[INFO] +- uk.gov.companieshouse:environment-reader-library:jar:1.3.10:compile
[INFO] |  \- commons-validator:commons-validator:jar:1.7:compile
[INFO] |     +- commons-beanutils:commons-beanutils:jar:1.9.4:compile
[INFO] |     +- commons-digester:commons-digester:jar:2.1:compile
[INFO] |     \- commons-collections:commons-collections:jar:3.2.2:compile
[INFO] +- commons-lang:commons-lang:jar:2.6:compile
[INFO] +- org.springframework.boot:spring-boot-starter-test:jar:3.3.1:test
[INFO] |  +- org.springframework.boot:spring-boot-test:jar:3.3.1:test
[INFO] |  +- org.springframework.boot:spring-boot-test-autoconfigure:jar:3.3.1:test
[INFO] |  +- com.jayway.jsonpath:json-path:jar:2.9.0:test
[INFO] |  +- jakarta.xml.bind:jakarta.xml.bind-api:jar:4.0.2:test
[INFO] |  |  \- jakarta.activation:jakarta.activation-api:jar:2.1.3:test
[INFO] |  +- net.minidev:json-smart:jar:2.5.1:test
[INFO] |  |  \- net.minidev:accessors-smart:jar:2.5.1:test
[INFO] |  |     \- org.ow2.asm:asm:jar:9.6:test
[INFO] |  +- org.assertj:assertj-core:jar:3.25.3:test
[INFO] |  |  \- net.bytebuddy:byte-buddy:jar:1.14.17:test
[INFO] |  +- org.awaitility:awaitility:jar:4.2.1:test
[INFO] |  +- org.hamcrest:hamcrest:jar:2.2:test
[INFO] |  +- org.junit.jupiter:junit-jupiter:jar:5.10.2:test
[INFO] |  |  \- org.junit.jupiter:junit-jupiter-params:jar:5.10.2:test
[INFO] |  +- org.mockito:mockito-core:jar:5.11.0:test
[INFO] |  |  +- net.bytebuddy:byte-buddy-agent:jar:1.14.17:test
[INFO] |  |  \- org.objenesis:objenesis:jar:3.3:test
[INFO] |  +- org.skyscreamer:jsonassert:jar:1.5.1:test
[INFO] |  |  \- com.vaadin.external.google:android-json:jar:0.0.20131108.vaadin1:test
[INFO] |  +- org.springframework:spring-test:jar:6.1.10:test
[INFO] |  \- org.xmlunit:xmlunit-core:jar:2.9.1:test
[INFO] +- org.junit.jupiter:junit-jupiter-engine:jar:5.10.2:test
[INFO] |  +- org.junit.platform:junit-platform-engine:jar:1.10.2:test
[INFO] |  |  +- org.opentest4j:opentest4j:jar:1.3.0:test
[INFO] |  |  \- org.junit.platform:junit-platform-commons:jar:1.10.2:test
[INFO] |  +- org.junit.jupiter:junit-jupiter-api:jar:5.10.2:test
[INFO] |  \- org.apiguardian:apiguardian-api:jar:1.1.2:test
[INFO] +- org.mockito:mockito-junit-jupiter:jar:5.11.0:test
[INFO] +- org.sonarsource.scanner.maven:sonar-maven-plugin:jar:3.10.0.2594:test
[INFO] |  +- org.codehaus.plexus:plexus-sec-dispatcher:jar:2.0:test
[INFO] |  |  +- org.codehaus.plexus:plexus-utils:jar:3.4.1:test
[INFO] |  |  +- org.codehaus.plexus:plexus-cipher:jar:2.0:test
[INFO] |  |  \- javax.inject:javax.inject:jar:1:test
[INFO] |  \- org.sonarsource.scanner.api:sonar-scanner-api:jar:2.16.3.1081:test
[INFO] +- org.apache.tomcat.embed:tomcat-embed-core:jar:9.0.89:compile
[INFO] +- org.apache.tomcat.embed:tomcat-embed-websocket:jar:9.0.89:compile
[INFO] |  \- org.apache.tomcat:tomcat-websocket-api:jar:9.0.89:compile
[INFO] +- org.apache.tomcat.embed:tomcat-embed-el:jar:9.0.89:compile
[INFO] +- org.apache.tomcat:tomcat-annotations-api:jar:9.0.89:compile
[INFO] +- com.fasterxml.jackson.core:jackson-databind:jar:2.15.3:compile
[INFO] +- com.fasterxml.jackson.core:jackson-annotations:jar:2.15.3:compile
[INFO] +- com.fasterxml.jackson.core:jackson-core:jar:2.15.3:compile
[INFO] +- com.fasterxml.jackson.datatype:jackson-datatype-jdk8:jar:2.15.3:compile
[INFO] +- com.fasterxml.jackson.module:jackson-module-parameter-names:jar:2.15.3:compile
[INFO] +- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:jar:2.15.3:compile
[INFO] \- jakarta.servlet:jakarta.servlet-api:jar:6.0.0:provided
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------

Reply:
Ok if it is not already included then we'll have to add it. But can you review the need of the `provided` scope?

Reply:
thanks @rafahop I have removed the scope attribute.